plugins/net_plugin/net_plugin.cpp

```diff
@@ -2008,7 +2008,7 @@ namespace eosio {
                  ("ex", -reversible_remaining)("fs", fork_db_size));
          reversible_remaining = 0;
       }
-      if (reversible_remaining < sync_fetch_span) {
+      if ((uint32_t)reversible_remaining < sync_fetch_span) {
```

Comment:
I usually don't like casting away signedness warnings like this but here we've established just above that `reversible_remaining >=0` so it's completely safe. Not sure what would be a better pattern in this case.

Reply:
I don't care for the cast either. Maybe something like:

```
      int32_t reversible_remaining = my_impl->chain_plug->chain().max_reversible_blocks_allowed();
      uint32_t constrained_reversible_remaining = 0;
      if (reversible_remaining <= 0) {
         auto fork_db_size = my_impl->chain_plug->chain().fork_db_size();
         fc_wlog(logger, "max-reversible-blocks exceeded by ${ex}, fork_db_size ${fs}",
                 ("ex", -reversible_remaining)("fs", fork_db_size));
      } else {
         constrained_reversible_remaining = reversible_remaining;
      }
      if (constrained_reversible_remaining < sync_fetch_span) {
         auto fork_db_size = my_impl->chain_plug->chain().fork_db_size();
         fc_wlog(logger, "sync-fetch-span ${sfs} restricted to ${r} by max-reversible-blocks, fork_db_size ${fs}",
                 ("sfs", sync_fetch_span)("r", constrained_reversible_remaining )("fs", fork_db_size));
         return constrained_reversible_remaining ;
      }
```

Reply:
Reminds me of programming in Ada.